tests/jballerina-unit-test/src/test/resources/test-src/endpoint/new/remote_basic.bal

```diff
@@ -96,3 +96,14 @@ function clientObjectDeclaredInIfStatementElseBlock() returns int {
         return x->action2("0", false);
     }
 }
+
+service class Foo {
+    remote function bar() {
+
+    }
+}
+
+public function testBoundMethodAccess() {
+    Foo foo = new;
+    any _ = foo.bar;
```

Comment:
Was wondering if we could actually assert something here and I now have a concern if we should even allow bound method access when the method is a `remote` method. 

What is the type set to `foo.bar`? We don't allow `remote` in function type descriptors, so we can't quite set a type for `foo.bar` without losing the info about it being `remote`. This should probably be disallowed, let's discuss.